Library/ViewModels/WithShippingRewardsCollectionViewModel.swift

```diff
@@ -429,5 +438,56 @@ private func backingAndShippingTotal(for project: Project, and reward: Reward) -
 private func allowableSortedProjectRewards(from project: Project) -> [Reward] {
   let availableRewards = project.rewards.filter { rewardIsAvailable($0) }
   let unAvailableRewards = project.rewards.filter { !rewardIsAvailable($0) }
+
   return availableRewards + unAvailableRewards
 }
+
+private func filteredRewardsByLocation(
+  _ rewards: [Reward],
+  shippingRule: ShippingRule?
+) -> [Reward] {
+  return rewards.filter { reward in
+    var shouldDisplayReward = false
+
+    let isRewardLocalOrDigital = isRewardDigital(reward) || isRewardLocalPickup(reward)
+    let isUnrestrictedShippingReward = reward.isUnRestrictedShippingPreference
+    let isRestrictedShippingReward = reward.isRestrictedShippingPreference
+
+    // return all rewards that are digital or ship anywhere in the world.
+    if isRewardLocalOrDigital || isUnrestrictedShippingReward {
+      shouldDisplayReward = true
+
+      // if add on is local for local base, ensure locations are equal before displaying
+      if isRewardLocalPickup(reward) {
+        if let rewardLocation = reward.localPickup?.country,
+           let shippingRuleLocation = shippingRule?.location.country, rewardLocation == shippingRuleLocation {
+          shouldDisplayReward = true
+        } else {
+          shouldDisplayReward = false
+        }
+      }
+      // If restricted shipping, compare against selected shipping location
+    } else if isRestrictedShippingReward {
+      shouldDisplayReward = rewardShipsTo(selectedLocation: shippingRule?.location.id, reward)
+    }
+
+    return shouldDisplayReward
+  }
+}
+
+/**
+ For base rewards that have restricted shipping, only return
+ add-ons that can ship to the selected shipping location.
+ */
+private func rewardShipsTo(
+  selectedLocation locationId: Int?,
+  _ reward: Reward
+) -> Bool {
+  guard let selectedLocationId = locationId else { return false }
+
+  let shippingLocationIds: Set<Int> = Set(
+    reward.shippingRulesExpanded?.map(\.location).map(\.id) ?? []
+  )
+
+  return shippingLocationIds.contains(selectedLocationId)
+}
```

Comment:
My current issue. This logic is taken from the Add-Ons screen that currently filters addons based on location, but I'm not sure how that really works because from what I can tell...

Rewards with restricted shipping don't have location or shipping rules associated with them; their values are `nil,` so I don't know how to compare them to the selected shipping location in the dropdown. Working on seeing what's available on the backend and if the team as any insights.